oracle_fdw.c

```diff
@@ -1565,16 +1570,40 @@ oraclePlanForeignModify(PlannerInfo *root, ModifyTable *plan, Index resultRelati
 	RTEPermissionInfo *perminfo = getRTEPermissionInfo(root->parse->rteperminfos, rte);
 
 	check_user = perminfo->checkAsUser;
-	updated_cols = perminfo->updatedCols;
+	if (root->parse->commandType != CMD_UPDATE)
+	{
+		updated_cols = bms_copy(perminfo->updatedCols);
+	}
 #else
 	check_user = rte->checkAsUser;
-#if PG_VERSION_NUM >= 90500
+#if (PG_VERSION_NUM >= 130010 && PG_VERSION_NUM < 140000) || \
+	(PG_VERSION_NUM >= 140007 && PG_VERSION_NUM < 150000) || \
+	(PG_VERSION_NUM >= 150002 && PG_VERSION_NUM < 160000)
```

Comment:
Can you explain these version checks?
Was there a compatibility break in a minor release?
In general, I don't cater for that. You should just let the build fail with the old minor release.
We can expect users to apply minor updates.
Besides, what happens if the user builds oracle_fdw with 15.2 and runs against 15.1?